### PR TITLE
feat(475): Redesign Landing Page for the Blog

### LINKED ISSUE #475: Redesign Landing Page for the Blog
Update the blog home page design as per the new design:

Blog Landing Page-V1

**Technical Requirements**
- Search bar isn't required at the moment.
- For "featured posts" use front matter to extract that information.
- Load more should load all the blogs.
- Tabs represent categories and should also be loaded from front matter.
- UI should be responsive.

### PR DESCRIPTION
**Pull request by neo773**
https://github.com/user-attachments/assets/2aa1d53e-1703-4f92-a529-5ce45ad648c4

/claim #475
/closes #475

### CODE REVIEW

**Comment by neo773** [comment]
@tusharmath 
Could I get a review on this?

**Comment by amitksingh1490** [comment]
image
Responsiveness is broken

**Comment by neo773** [comment]
@amitksingh1490 
Could you check again? updated mobile design to match Figma file it was using Desktop layout before.

**Comment by neo773** [comment]
@amitksingh1490

File: src/theme/BlogCategories/index.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+  return (
+    <div className="mb-4 flex items-center space-x-4 border-b border-gray-200">
+      {Object.entries(categories).map(([name, count]) => (
+        <div
+          aria-role="button"
+          aria-label={`${name} (${count})`}
+          key={name}
+          onClick={() => onCategoryClick(name === activeCategory ? "All" : name)}
+          style={{
```
Comment (amitksingh1490) [inline src/theme/BlogCategories/index.tsx]: no Inline styles

**Review by amitksingh1490 — CHANGES_REQUESTED** [review]

File: blog/2023-graphql-conf-2023-09-29.md [hunk 2]
```diff
@@ -9,6 +9,7 @@ hide_table_of_contents: true
 slug: graphql-conf-2023
 canonical_url: https://tailcall.hashnode.dev/graphql-conf-2023
 category: Tutorial
+featured: true
```
Comment (amitksingh1490) [inline blog/2023-graphql-conf-2023-09-29.md]: don't add featured will add different ones

File: blog/api-orchestration-2023-06-12.md [hunk 3]
```diff
@@ -11,6 +11,7 @@ image: /images/blog/bff-architecture.png
 hide_table_of_contents: true
 slug: no-one-talks-about-api-orchestration
 canonical_url: https://tailcall.hashnode.dev/no-one-talks-about-api-orchestration
+featured: true
```
Comment (amitksingh1490) [inline blog/api-orchestration-2023-06-12.md]: don't add featured will add different ones

Comment (amitksingh1490) [reply blog/api-orchestration-2023-06-12.md]: Its still coming as featured post
image

**Review by amitksingh1490 — CHANGES_REQUESTED** [review]

**Comment by amitksingh1490** [comment]
image
design is not matching for web as well as mobile layout
marking as draft

**Comment by hunxjunedo** [comment]
@neo773 it would be my pleasure collaborating with you on this one.

**Comment by neo773** [comment]
> @neo773 it would be my pleasure collaborating with you on this one.

I mean it's a done PR I'm not sure what's there to collaborate at this stage.

**Comment by amitksingh1490** [comment]
image (13)
Still design is not matching, moving this to draft

**Comment by neo773** [comment]
@amitksingh1490 
About the tags, when I spoke with Tushar he wanted a separate `category` field in the frontmatter instead of using tags.
I can change it back let me know.

Some of the images have a white border the layout itself is fine

image

As for spacing pushed a new commit

**Comment by amitksingh1490** [comment]
What about featured post still coming even though nothing is marked as featured.

**Comment by neo773** [comment]
> What about featured post still coming even though nothing is marked as featured.

Fixed, I assumed we'll always have one featured post since the Figma file didn't have an empty state UI

File: docusaurus.config.ts [hunk 4]
```diff
@@ -239,7 +239,7 @@ export default {
         routeBasePath: "blog",
         include: ["**/*.{md,mdx}"],
         exclude: ["**/_*.{js,jsx,ts,tsx,md,mdx}", "**/_*/**", "**/*.test.{js,jsx,ts,tsx}", "**/__tests__/**"],
-        postsPerPage: 10,
+        postsPerPage: 999,
```
Comment (hunxjunedo) [inline docusaurus.config.ts]: Use 'ALL' if you want to get all posts at once.

Comment (neo773) [reply docusaurus.config.ts]: Resolved, Thanks.

**Comment by amitksingh1490** [comment]
image
image
Image getting cropped. 

image
featured section is visible in the tab even if there is no space for it.

**Comment by neo773** [comment]
changed images to `object-contain` but these images will need to replaced in my opinion they have no consistency in dimensions.

**Review by amitksingh1490 — APPROVED** [review]

**Pull request merged**